File: trustdevice/src/main/java/cn/tongdun/mobrisk/providers/risk/RiskInfoRawData.java
package cn.tongdun.mobrisk.providers.risk;

import android.util.Pair;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

import cn.tongdun.mobrisk.providers.RawData;

/**
 * @description: RiskInfoRawData
 * @author: wuzuchang
 * @date: 2022/12/6
 */
@Deprecated(since = "pro no such class")
public class RiskInfoRawData extends RawData<String> {
    private String root;
    private String debug;
    private String multiple;
    private String xposed;

    public RiskInfoRawData(String root, String debug, String multiple, String xposed) {
        this.root = root;
        this.debug = debug;
        this.multiple = multiple;
        this.xposed = xposed;
    }

    private String getRoot() {
        return root;
    }

    private String getDebug() {
        return debug;
    }

    private String getMultiple() {
        return multiple;
    }

    private String getXposedStatus() {
        return xposed;
    }

    @Override
    public List<Pair<String, String>> loadData() {
        List<Pair<String, String>> data = new ArrayList<>();
        data.add(new Pair<>("root", getRoot()));
        data.add(new Pair<>("debug", getDebug()));
        data.add(new Pair<>("multiple", getMultiple()));
        data.add(new Pair<>("xposed", getXposedStatus()));
        return data;
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/providers/RiskInfoRawData.kt
package cn.tongdun.mobrisk.providers

import cn.tongdun.mobrisk.core.tools.Constants

/**
 * @description:RiskInfoRawData
 * @author: wuzuchang
 * @date: 2023/5/16
 */
data class RiskInfoRawData(
    val root: Boolean,
    val debug: Boolean,
    val multiple: Boolean,
    val xposed: Boolean,
    val magisk: Boolean,
    val hook: Boolean,
    val emulator: Boolean,
    val vpn: Boolean,
    val deviceInfoTampered: Boolean
) {

    override fun toString(): String {
        val risks = arrayOf(
            root.takeIf { it }?.let { Constants.KEY_ROOT },
            debug.takeIf { it }?.let { Constants.KEY_DEBUG },
            multiple.takeIf { it }?.let { Constants.KEY_MULTIPLE },
            xposed.takeIf { it }?.let { Constants.KEY_XPOSED },
            magisk.takeIf { it }?.let { Constants.KEY_MAGISK },
            hook.takeIf { it }?.let { Constants.KEY_HOOK },
            emulator.takeIf { it }?.let { Constants.KEY_EMULATOR },
            vpn.takeIf { it }?.let { Constants.KEY_VPN },
            deviceInfoTampered.takeIf { it }?.let { Constants.KEY_DEVICE_INFO_TAMPERED }
        )
        return risks.filterNotNull().joinToString()
    }
}